@here/olp-sdk-dataservice-read/lib/utils/QuadKeyUtils.ts

```diff
@@ -217,3 +217,89 @@ export function isValid(key: QuadKey): boolean {
     const columnValid = key.column >= 0 && key.column < dimensionAtLevel;
     return rowValid && columnValid;
 }
+
+/**
+ * Converts the QuadKey object into a string.
+ *
+ * If the tile is the root tile, the quad string is '-'. Otherwise the string is a number to the
+ * base of 4, but without the leading 1, with the following properties:
+ *  1. the number of digits equals the level.
+ *  2. removing the last digit gives the parent tile's quadkey string, i.e. appending 0,1,2,3
+ *     to a quadkey string gives the tiles's children.
+ */
+export function quadKeytoQuadString(key: QuadKey): string {
+    let result = "";
+
+    for (let i = key.level; i > 0; --i) {
+        // tslint:disable:no-bitwise
+        const mask = 1 << (i - 1);
+
+        const col = (key.column & mask) !== 0;
+        const row = (key.row & mask) !== 0;
+
+        if (col && row) {
+            result += "3";
+        } else if (row) {
+            result += "2";
+        } else if (col) {
+            result += "1";
+        } else {
+            result += "0";
+        }
+        // tslint:enable:no-bitwise
+    }
+
+    return result;
+}
+
+/**
+ * Creates a QuadKey object key from a quad string.
+ *
+ * @param key The string, represented  quadkey number in base 4 to convert.
+ * @returns An object, implemented QuadKey interface.
+ */
+export function quadKeyfromQuadString(key: string): QuadKey {
+    const level = key.length;
+    let row = 0;
+    let column = 0;
+    for (let i = 0; i < key.length; ++i) {
+        // tslint:disable:no-bitwise
+        const mask = 1 << i;
+        const d = parseInt(key.charAt(level - i - 1), 10);
+        if (d & 0x1) {
+            column |= mask;
+        }
+        if (d & 0x2) {
+            row |= mask;
+        }
+        // tslint:enable:no-bitwise
+    }
+    return { row, column, level };
+}
+
+/**
+ * Returns a sub quad key that is relative to its parent.
+ *
+ * This function can be used to generate sub keys that are relative to a parent that is delta
+ * levels up in the quadtree.
+ *
+ * This function can be used to create shortened keys for quads on lower levels if the parent is
+ * known.
+ *
+ * Note - the sub quadkeys fit in a 16-bit unsigned integer if the `delta` is smaller than 8. If
+ * `delta` is smaller than 16, the sub quadkey fits into an unsigned 32-bit integer.
+ *
+ * Deltas larger than 16 are not supported.
+ *
+ * @param delta The number of levels relative to its parent quadkey. Must be greater or equal to
+ * 0 and smaller than 16.
+ * @returns The quadkey string relative to its parent that is `delta` levels up the tree.
+ */
+export function getRelatedSubQuadKey(code: number, delta: number): string {
```

Comment:
Is this function similar to C++ GetSubtile()?
Then I would not do it like this but rather work on the QuadKey also as a method -> `QuadKey.getSubKey(delta)`

Reply:
yes, It's similar.  I will create a TileKey class because we do not have it. We have only the QuadKey interface

Reply:
Added TileKey class, the similar to harp.gl